llvm/lib/CodeGen/SelectionDAG/DAGCombiner.cpp

```diff
@@ -9893,7 +9894,8 @@ SDValue DAGCombiner::visitSHL(SDNode *N) {
     // folding this will increase the total number of instructions.
     if (N0.getOpcode() == ISD::SRL &&
         (N0.getOperand(1) == N1 || N0.hasOneUse()) &&
-        TLI.shouldFoldConstantShiftPairToMask(N, Level)) {
+        TLI.shouldFoldConstantShiftPairToMask(N, Level) &&
+        !isCanBeLoadedWithLsl(N)) {
```

Comment:
Why not move the call to `isCanBeLoadedWithLsl()` outside of `SelectionDAG` and into the `aarch64` target?